src/screen/dashboard/sidebar.rs

```diff
@@ -61,39 +60,176 @@ impl Sidebar {
     pub fn new() -> Self {
         Self {
             hidden: false,
-            timestamp: None,
+            reloading_config: false,
         }
     }
 
     pub fn toggle_visibility(&mut self) {
         self.hidden = !self.hidden
     }
 
-    pub fn update(&mut self, message: Message) -> Event {
+    pub fn update(&mut self, message: Message) -> Option<Event> {
         match message {
-            Message::Open(source) => Event::Open(source),
-            Message::Popout(source) => Event::Popout(source),
-            Message::Focus(window, pane) => Event::Focus(window, pane),
-            Message::Replace(window, source, pane) => Event::Replace(window, source, pane),
-            Message::Close(window, pane) => Event::Close(window, pane),
+            Message::Open(source) => Some(Event::Open(source)),
+            Message::Popout(source) => Some(Event::Popout(source)),
+            Message::Focus(window, pane) => Some(Event::Focus(window, pane)),
+            Message::Replace(window, source, pane) => Some(Event::Replace(window, source, pane)),
+            Message::Close(window, pane) => Some(Event::Close(window, pane)),
             Message::Swap(from_window, from_pane, to_window, to_pane) => {
-                Event::Swap(from_window, from_pane, to_window, to_pane)
+                Some(Event::Swap(from_window, from_pane, to_window, to_pane))
             }
-            Message::Leave(buffer) => Event::Leave(buffer),
-            Message::ToggleFileTransfers => Event::ToggleFileTransfers,
-            Message::ToggleCommandBar => Event::ToggleCommandBar,
-            Message::ToggleThemeEditor => Event::ToggleThemeEditor,
+            Message::Leave(buffer) => Some(Event::Leave(buffer)),
+            Message::ToggleFileTransfers => Some(Event::ToggleFileTransfers),
+            Message::ToggleCommandBar => Some(Event::ToggleCommandBar),
+            Message::ToggleThemeEditor => Some(Event::ToggleThemeEditor),
             Message::ReloadConfigFile => {
-                self.timestamp = Some(Instant::now());
-                Event::ReloadConfigFile
+                self.reloading_config = true;
+                Some(Event::ReloadConfigFile)
+            },
+            Message::OpenReleaseWebsite => Some(Event::OpenReleaseWebsite),
+            Message::ReloadComplete => {
+                self.reloading_config = false;
+                None
             }
```

Comment:
Since we have some "complex" interaction here w/ our widget state being tied to the config being loaded, I think it actually makes more sense to call `Task::perform(Config::load(), Message)` at this level, then return an `Event::ConfigReloaded(config)`.

That will allow us to:

- Directly handle the icon state of loaded / error -> reset back to normal icon as we do in `theme_editor` (checkout how `Message::Saved(_)` is handled and how we can chain a `time::sleep` to automatically reset the icon back to normal)
- We won't have to "know" about calling `dashboard.config_reloaded() -> sidebar.config_reloaded()` to handle this state if we're instead reloading from `Hallow / main.rs`

Reply:
let me know what you think about the current iteration. All state manipulation for the sidebar is done in `sidebar.rs` (which is much cleaner, thank you for the suggestion). However, config reloading from commands and shortcuts happens in `dashboard.rs`.